sympy/tensor/tests/test_indexed.py

```diff
@@ -353,3 +353,12 @@ def test_indexed_is_constant():
     assert not A[1+2*i, k].is_constant(i)
     assert A[1+2*i, k].is_constant(j)
     assert not A[1+2*i, k].is_constant(k)
+
+
+def test_issue_12533():
+    d = IndexedBase('d')
+    assert IndexedBase(range(5)) == Array([0, 1, 2, 3, 4])
+    assert d[0].subs(Symbol("d"), range(5)) == 0
```

Comment:
(Continuing from https://github.com/sympy/sympy/pull/12545#discussion_r111691858)

I see now that 

```
In [31]: IndexedBase('d').args
Out[31]: (d,)
```

It doesn't really make sense to me that you would substitute `d`, or really the line https://github.com/sympy/sympy/pull/12556/files#diff-f91a3a19a7672c48a33a63796c8e9b09R380

Everything still works if you replace `d` directly instead of its arg

```
In [10]: d = IndexedBase('d')

In [11]: d[Idx('i')].subs(d, (1, 2, 3))
Out[11]: (1, 2, 3)[i]
```

By the way, you should add some tests where the index value is symbolic.